main.tf

```diff
@@ -2709,6 +2709,160 @@ module "vpa" {
   tags = var.tags
 }
 
+
+################################################################################
+# Velero
+################################################################################
+locals {
+  velero_name            = "velero"
+  velero_service_account = try(var.velero.service_account_name, "${local.velero_name}-sa")
+}
+
+# https://github.com/vmware-tanzu/velero-plugin-for-aws#option-1-set-permissions-with-an-iam-user
+data "aws_iam_policy_document" "velero" {
+  statement {
+    actions = [
+      "ec2:CreateSnapshot",
+      "ec2:CreateSnapshots",
+      "ec2:CreateTags",
+      "ec2:CreateVolume",
+      "ec2:DeleteSnapshot"
+    ]
+    resources = [
+      "arn:${local.partition}:ec2:${local.region}:${local.account_id}:instance/*",
+      "arn:${local.partition}:ec2:${local.region}::snapshot/*",
+      "arn:${local.partition}:ec2:${local.region}:${local.account_id}:volume/*"
+    ]
+  }
+
+  statement {
+    actions = [
+      "ec2:DescribeSnapshots",
+      "ec2:DescribeVolumes"
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "s3:AbortMultipartUpload",
+      "s3:DeleteObject",
+      "s3:GetObject",
+      "s3:ListMultipartUploadParts",
+      "s3:PutObject",
+    ]
+    resources = ["arn:${local.partition}:s3:::${var.velero.backup_s3_bucket}/*"]
+  }
+
+  statement {
+    actions   = ["s3:ListBucket"]
+    resources = ["arn:${local.partition}:s3:::${var.velero.backup_s3_bucket}"]
+  }
+}
+
+module "velero" {
+  # source = "aws-ia/eks-blueprints-addon/aws"
+  source = "./modules/eks-blueprints-addon"
+
+  create = var.enable_velero
+
+  # https://github.com/vmware-tanzu/helm-charts/blob/main/charts/velero/Chart.yaml
+  name             = try(var.velero.name, local.velero_name)
+  description      = try(var.velero.description, "A Helm chart to install the Velero")
+  namespace        = try(var.velero.namespace, "velero")
+  create_namespace = try(var.velero.create_namespace, true)
+  chart            = "velero"
+  chart_version    = try(var.velero.chart_version, "3.1.6")
+  repository       = try(var.velero.repository, "https://vmware-tanzu.github.io/helm-charts/")
+  values = try(var.velero.values, [<<-EOT
+  initContainers:
```

Comment:
Anything that's required should go into set so that users don't need to re-provide all of these if they want to provide values

Reply:
True. I’ll fix this, should be in the set.